src/main/kotlin/com/zopa/ktor/opentracing/OpenTracingServer.kt

```diff
@@ -21,22 +21,28 @@ import java.util.Stack
 
 
 class OpenTracingServer(
-        val filters: List<(ApplicationCall) -> Boolean>
+        val filters: List<(ApplicationCall) -> Boolean>,
+        val tagsToExtractFromPath: Map<String, Regex>
 ) {
     class Configuration {
         val filters = mutableListOf<(ApplicationCall) -> Boolean>()
+        val tagsToExtractFromPath: MutableMap<String, Regex> = mutableMapOf(uuidTagAndReplace)
 
         fun filter(predicate: (ApplicationCall) -> Boolean) {
             filters.add(predicate)
         }
+
+        fun extractTagFromPath(tagName: String, regex: Regex) {
+            tagsToExtractFromPath[tagName] = regex
```

Comment:
minor; can we have the regex as the first parameter? Seems more idiomatic , as in String.replace

Reply:
Was trying to be consistent throughout the code `extractTagFromPath(tagName, regex)` -> `Pair(tagName, regex)` -> `Pair(tagName, tagValue)` -> `spanBuilder.withTag(tagName, tagValue)`

Reply:
I get you but the implementation shouldn't be influenced by what's on the interface here.

Reply:
Yeah fair enough. I guess I was thinking of it as primarily a tag to be added, and you were thinking of it as primarily a replacement in the path 😁 I will switch it around